### PR TITLE
Arc: Improvements to search tabs and spaces

### PR DESCRIPTION
**Pull request by thomaspaulmann**
## Description

This is an update to the Arc extension that allows searching for tabs and spaces. It also incorporates new functionality like opening a tab in a specific space. Additionally, it opens links in already open tabs. This builds on top of Arc's new AppleScript API.

## Screencast

New screenshot was added to the metadata

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by yug2005** [comment]
Hey I am getting the following error with `Search Tabs`: 
```
Error: Command failed with exit code 1: osascript -e 
    on escape_value(this_text)
      set AppleScript's text item delimiters to the "\""
      set the item_list to every text item of this_text
      set AppleScript's text item delimiters to the "\\\""
      set this_text to the item_list as string
      set AppleScript's text item delimiters to ""
      return this_text
    end replace_chars

    set _output to ""

    tell application "Arc"
      set _window_index to 1
      
      repeat with _window in windows
        set _tab_index to 1
        
        repeat with _tab in tabs of _window
          set _title to my escape_value(get title of _tab)
          set _url to get URL of _tab
          set _location to get location of _tab
          
          set _output to (_output & "{ \"title\": \"" & _title & "\", \"url\": \"" & _url & "\", \"windowId\": " & _window_index & ", \"tabId\": " & _tab_index & " , \"location\": \"" & _location & "\" }")
          
          if _tab_index < (count tabs of _window) and _window_index < 3 then
            set _output to (_output & ",\n")
          else
            set _output to (_output & "\n")
          end if
          
          set _tab_index to _tab_index + 1
        end repeat
        
        set _window_index to _window_index + 1
      end repeat
    end tell
    
    return "[\n" & _output & "\n]"
   
680:688: execution error: Arc got an error: Can’t make location of item 1 of every tab of item 1 of every window into type specifier. (-1700)
    at makeError (/Users/yug/.config/raycast/extensions/arc/search-tabs.js:11448:17)
    at handlePromise (/Users/yug/.config/raycast/extensions/arc/search-tabs.js:12019:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runAppleScript (/Users/yug/.config/raycast/extensions/arc/search-tabs.js:14530:22)
    at async Object.getTabs [as current] (/Users/yug/.config/raycast/extensions/arc/search-tabs.js:14536:20)
```
and the following error with `Search Spaces`: 
```
Error: Command failed with exit code 1: osascript -e 
    set _output to ""

    tell application "Arc"    
      set _space_index to 1  
      
      repeat with _space in spaces of front window
        set _title to get title of _space
        
        set _output to (_output & "{ \"title\": \"" & _title & "\", \"id\": " & _space_index & " }")
        
        if _space_index < (count spaces of front window) then
          set _output to (_output & ",\n")
        else
          set _output to (_output & "\n")
        end if
        
        set _space_index to _space_index + 1
      end repeat
    end tell
    
    return "[\n" & _output & "\n]"
   
98:549: execution error: Arc got an error: Can’t make spaces of window 1 into type specifier. (-1700)
    at makeError (/Users/yug/.config/raycast/extensions/arc/search-spaces.js:6010:17)
    at handlePromise (/Users/yug/.config/raycast/extensions/arc/search-spaces.js:6581:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runAppleScript (/Users/yug/.config/raycast/extensions/arc/search-spaces.js:9091:22)
    at async Object.getSpaces [as current] (/Users/yug/.config/raycast/extensions/arc/search-spaces.js:9108:20)
```

**Review by textnav — APPROVED** [review]
LGTM

File: extensions/arc/src/sql.ts [hunk 1]
```diff
@@ -73,8 +73,6 @@ export const useSQL = <Result>(path: string, query: string) => {
       } catch (e) {
         console.log("type", typeof e);
         if (error instanceof Error && error.message.includes("operation not permitted")) {
-          console.log("ERRRORROROR");
-
           setError(new PermissionError("You do not have permission to access the History database."));
         } else {
           console.log("asdfasdf");
```
[the remark this answers is not in the document]
Comment (thomaspaulmann) [reply extensions/arc/src/sql.ts:78]: Haha yeah removing it.

**Comment by thomaspaulmann** [comment]
@yug2005 do you have the latest update installed?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**